### PR TITLE
Disable access to Alertmanager (user, or sending alerts), if it is not running.

### PR DESCRIPTION
**Pull request by pstibrany**
**What this PR does**: This PR disables access to user's Alertmanager (by user, or services trying to send alert) until `MultitenantAlertmanager` has started and synced configuration for each user.

Call to tenant's AM before `MultitenantAlertmanager` is running may result in loss of configuration for this tenant.

**Checklist**
- [x] Tests updated
- [na] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by gouthamve — DISMISSED** [review]

**Review by jtlisi — COMMENTED** [review]
Is this enough? In addition, shouldn't we require a check to the backend before storing the fallback? Othwerwise, we could end up in a similar situation in the future. For instance, once we start sharding Alertmanager tenants if a request somehow gets routed to an Alertmanager without the tenant loaded it will override the config.

**Comment by pstibrany** [comment]
> Is this enough?

I think yes, for now.

> In addition, shouldn't we require a check to the backend before storing the fallback? Othwerwise, we could end up in a similar situation in the future. For instance, once we start sharding Alertmanager tenants if a request somehow gets routed to an Alertmanager without the tenant loaded it will override the config.

That's a valid point, and something keep an eye on when reviewing AM sharding. I think we should also reconsider idea of uploading "empty" config to the same path as user-uploaded config. Eg. we could give empty config ".default" suffix.

**Review by gotjosh — APPROVED** [review]
Good catch @pstibrany !

File: CHANGELOG.md [hunk 1]
```diff
@@ -44,6 +44,7 @@
 * [BUGFIX] Compactor: delete `deletion-mark.json` at last when deleting a block in order to not leave partial blocks without deletion mark in the bucket if the compactor is interrupted while deleting a block. #3660
 * [BUGFIX] Blocks storage: do not cleanup a partially uploaded block when `meta.json` upload fails. Despite failure to upload `meta.json`, this file may in some cases still appear in the bucket later. By skipping early cleanup, we avoid having corrupted blocks in the storage. #3660
 * [BUGFIX] Alertmanager: disable access to `/alertmanager/metrics` (which exposes all Cortex metrics), `/alertmanager/-/reload` and `/alertmanager/debug/*`, which were available to any authenticated user with enabled AlertManager. #3678
+* [BUGFIX] Alertmanager: don't serve HTTP requests until AlertManager has fully started. Serving HTTP requests earlier may result in loss of configuration for the user. #3679
```
Comment (gotjosh) [inline CHANGELOG.md:47]:
```suggestion
* [BUGFIX] Alertmanager: don't serve HTTP requests until Alertmanager has fully started. Serving HTTP requests earlier may result in loss of configuration for the user. #3679
```

**Review by jtlisi — APPROVED** [review]
LGTM modulo changelog suggestion

**Review by pracucci — APPROVED** [review]
LGTM!

**Pull request merged**